base/inc/AdePT/BVHNavigator.h

```diff
@@ -197,7 +197,7 @@ class BVHNavigator {
       }
 
       if (out_state.Top() != nullptr) {
-        while (out_state.Top()->IsAssembly()) {
+        while (out_state.Top()->IsAssembly() || out_state.GetNavIndex() == in_state.GetNavIndex()) {
```

Comment:
What's the reason for this? Isn't this just hiding other problems?

Reply:
This method has the contract to cross into a different location, which in rare cases may not happen because relocation is done using the inverse transformation so a mismatch may appear making the algorithm find again the exited volume, but on the wrong side of the boundary (returning negative distance for DistanceToIn, converted to zero and making a looper). I would claim that the modification hides indeed the real fix, which would be to use the direct transformation also for relocation, but allows to recover the looping error w/o introducing a bias.